pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1600,7 +1671,14 @@ public void closeFailed(ManagedLedgerException exception, Object ctx) {
             closeFuture.completeExceptionally(exception);
             return null;
         });
-
+        if (!disconnectClients) {
+            FutureUtil.completeAfter(closeFutures.notDisconnectClients, closeFuture);
+        } else {
+            FutureUtil.completeAfter(closeFutures.notDisconnectClients, closeFuture);
+            FutureUtil.completeAfter(closeFutures.notWaitDisconnectClients, closeFuture);
```

Comment:
Is there ever a difference between `notDisconnectClients` and `notWaitDisconnectClients`?

Reply:
- call `close(true /* disconnectClients */, true /* closeWithoutWaitingClientDisconnect */)` will return `notWaitDisconnectClients` cached.
- call `close(false /* disconnectClients */)` will return `notDisconnectClients` cached

See more details at the description of the internal class `CloseFutures` [here](https://github.com/apache/pulsar/pull/17524/files#diff-5edf14cc6f25857d0cfdd26b2d3b3141230ecfb0dfa95aebf7583fd76ede4c4bR275-R288)

Reply:
Yes I noticed that, but is the actual state of the fields ever different?

Reply:
I think it's fine. I guess it's a modeling question. :)

Reply:
If the first closing was called `close(false /* disconnectClients */)`, the both fields will be different, otherwise they are the same.